src/python/grpcio/grpc/_interceptor.py

```diff
@@ -547,7 +587,8 @@ def close(self):
         self._channel.close()
 
 
-def intercept_channel(channel, *interceptors):
+def intercept_channel(channel: grpc.Channel,
+                      *interceptors: Optional[Sequence[Any]]) -> grpc.Channel:
```

Comment:
I think the type of `*interceptors` should be `Union[UnaryUnaryClientInterceptor, ..., StreamStreamClientInterceptor]`. [Splat arguments should be annotated as if they were a single argument](https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values).

Reply:
[You mean this check?](https://github.com/grpc/grpc/blob/d3d2b64d234337ed67564ca7b0eb1a15871b939b/src/python/grpcio/grpc/_interceptor.py#L551) Not everyone uses type annotations. Technically speaking this would be a regression, so I'm leaning away from that.